### PR TITLE
Use packages installed via Travis Homebrew addon [ECR-3951]

### PR DESCRIPTION
**Pull request by vitvakatu**
## Overview

---
See: https://jira.bf.local/browse/ECR-XYZ

### Definition of Done

- [ ] There are no TODOs left in the code
- [ ] Change is covered by automated [tests](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#tests)
- [ ] The [coding guidelines](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [ ] Public API has Javadoc
- [ ] Method preconditions are checked and documented in the Javadoc of the method
- [ ] Changelog is updated if needed (in case of notable or breaking changes)
- [ ] The continuous integration build passes

### CODE REVIEW

**Comment by dmitry-timofeev** [comment]
>     e23d32a Use latest xcode version

AFAIK, java version is tied to xcode version. It is only possible to explicitly override that version by using a script (which will download it) — but see their documentation.

**Comment by vitvakatu** [comment]
Yeah, I hoped it will work, according to this part of docs:

> While Mac jobs can test against multiple JDK versions using the jdk key, macOS images up to xcode9.3 can only switch up to Java 8, and images xcode9.4 and later can switch to Java 10 (if pre-installed) and later. In practical terms, if your Mac build requires Java 8 and below, use xcode9.3 (or below); if your build requires Java 10 and later, use xcode9.4 (or later).

Seems like it won't, so we can't update used brew version automatically, so we can't use Travis Homebrew addon (see https://github.com/Homebrew/homebrew-bundle/issues/540). I'm afraid we won't be able to speed up builds on Mac that way.

**Comment by dmitry-timofeev** [comment]
Do I understand correctly that the brew snapshot is tied to xcode version and is not updated regularly (i.e., once xcode version is added, the brew snapshot is frozen — I wonder, though, how it works if brew does not keep the previous versions of artifacts — we bumped into such issue with rocksdb, I believe)? If that's correct, will using a recent xcode + jdk_switcher (or install_jdk script) help (i.e., is it faster to download JDK 8 on recent xcode than to update homebrew on the old xcode)?

**Comment by vitvakatu** [comment]
> Do I understand correctly that the brew snapshot is tied to xcode version and is not updated regularly

Not exactly, but close to truth. Travis uses `brew bundle` tool to download fresh brew snapshot, but this tool does not support brew version installed by default on used osx version. The fix is pretty simple: to run `brew update`, but this is exactly what we're hoping to get rid of.

**Comment by vitvakatu** [comment]
Nah, looks like switching jdk is entirely unsupported on Mac.

**Comment by vitvakatu** [comment]
@dmitry-timofeev installing JDK 8 from archive works, takes only 3.5 seconds, therefore Mac build takes only 18 minutes instead of 26.

**Comment by vitvakatu** [comment]
It is impossible to use `install-jdk.sh` script - it does not support JDK 8, only 9+

File: .travis.yml [hunk 1]
```diff
@@ -74,15 +77,21 @@ before_cache:
 before_install:
   # Skip the build if only md files were updated.
   - source .travis/skip-ci.sh
+  # Install JDK 8 on Mac
+  - |
+    if [[ "$TRAVIS_OS_NAME" == "osx" ]]
+    then wget "https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/download/jdk8u232-b09/OpenJDK8U-jdk_x64_mac_hotspot_8u232b09.tar.gz"
+    tar -xf OpenJDK8U-jdk_x64_mac_hotspot_8u232b09.tar.gz
+    export PATH=$PWD/jdk8u232-b09/Contents/Home/bin:$PATH
+    export JAVA_HOME=$PWD/jdk8u232-b09/Contents/Home/jre
```
Comment (dmitry-timofeev) [inline .travis.yml]: JAVA_HOME must be set to its home, not to jre.

File: .travis.yml [hunk 2]
```diff
@@ -74,15 +77,21 @@ before_cache:
 before_install:
   # Skip the build if only md files were updated.
   - source .travis/skip-ci.sh
+  # Install JDK 8 on Mac
```
Comment (dmitry-timofeev) [inline .travis.yml]: I'd document the 'why' and 'why not' for future us :-)

**Comment by dmitry-timofeev** [comment]
> takes only 3.5 seconds, therefore Mac build takes only 18 minutes instead of 26.

That's awesome 🎉

File: .travis.yml [hunk 3]
```diff
@@ -74,15 +77,23 @@ before_cache:
 before_install:
   # Skip the build if only md files were updated.
   - source .travis/skip-ci.sh
+  # Install JDK 8 on Mac. We use latest osx_image for faster Homebrew packages
+  # installation, but it has JDK 13 by default. We can't use Homebrew for installing
+  # a cask, because it will force Homebrew update, which takes a long time.
+  - |
+    if [[ "$TRAVIS_OS_NAME" == "osx" ]]
```
[the remark this answers is not in the document]
Comment (dmitry-timofeev) [reply .travis.yml]: It makes sense to test on both — but I don't think we need that 🙃 

8 is currently the minimum required, and _the one with which we build releases_, so it must work. We don't run _system_ tests on Mac internally, only a release script, therefore, if it breaks, we only notice during the release — and it blocks the release. On the other hand, whilst it is possible to have green Linux/8 and Mac/13 and red Mac/8 (native stuff could bring surprises), I don't think it is likely.

[the remark this answers is not in the document]
Comment (vitvakatu) [reply .travis.yml]: I think it's relatively simple approach to get Java 8 - so it makes sense to test on 8, as on a minimal supported version. So I'm leaving it as it is now.

**Pull request merged**